quickwit/quickwit-doc-mapper/src/query_builder.rs

```diff
@@ -33,38 +34,103 @@ pub(crate) fn build_query(
+            } else {
+                resolve_fields(&schema, &request.search_fields)?
+            };
+
+            if let Some(sort_by_field) = request.sort_by_field.as_ref() {
+                validate_sort_by_field_name(sort_by_field, &schema, Some(&search_fields))?;
+            }
 
-    let search_fields = if request.search_fields.is_empty() {
-        resolve_fields(&schema, default_field_names)?
-    } else {
-        resolve_fields(&schema, &request.search_fields)?
-    };
+            let mut query_parser =
+                QueryParser::new(schema, search_fields, QUICKWIT_TOKENIZER_MANAGER.clone());
+            query_parser.set_conjunction_by_default();
+            query_parser.parse_query(query)?
+        }
+        Some(SearchQuery::SetQuery(set_query)) => {
+            let _ = (&set_query.terms, &set_query.field_name, &set_query.tags);
+            let field_name = &set_query.field_name;
+            let field = schema
+                .get_field(field_name)
+                .ok_or_else(|| TantivyQueryParserError::FieldDoesNotExist(field_name.clone()))?;
+
+            let field_type = schema.get_field_entry(field).field_type();
+            if !field_type.is_indexed() {
+                return Err(anyhow::anyhow!("Attempted to search on not indexed field.").into());
+            }
 
-    if let Some(sort_by_field) = request.sort_by_field.as_ref() {
-        validate_sort_by_field_name(sort_by_field, &schema, Some(&search_fields))?;
-    }
+            // TODO maybe Facet could be allowed?
+            if matches!(field_type.value_type(), Type::Json | Type::Facet) {
+                return Err(
+                    anyhow::anyhow!("Attempted to search on unsuported field type.").into(),
+                );
+            }
+
+            // TODO error on any parse error instead??
+            let terms: Vec<_> = set_query
+                .terms
+                .iter()
+                // TODO not correct
+                .flat_map(|term| field_type.value_from_json(term.clone().into()).ok())
+                .map(|term| match term {
+                    Value::Str(text) => Term::from_field_text(field, &text),
+                    Value::PreTokStr(pre_tokenized_string) => {
+                        Term::from_field_text(field, &pre_tokenized_string.text)
+                    }
+                    Value::U64(int) => Term::from_field_u64(field, int),
+                    Value::I64(int) => Term::from_field_i64(field, int),
+                    Value::F64(int) => Term::from_field_f64(field, int),
+                    Value::Bool(b) => Term::from_field_bool(field, b),
+                    Value::Date(date) => Term::from_field_date(field, date),
+                    Value::Bytes(buf) => Term::from_field_bytes(field, &buf),
+                    Value::IpAddr(ip) => Term::from_field_ip_addr(field, ip),
+                    Value::JsonObject(_) | Value::Facet(_) => unreachable!(),
```

Comment:
I agree this is unreachable, but I'd rather return an error here too.

The point is to not make anyone think when thinking can this function panic.
```suggestion
                    Value::JsonObject(_) | Value::Facet(_) => unreachable!(),
```